examples/DrawingInTheAir/DrawingInTheAir_MotionActivated

```diff
@@ -0,0 +1,290 @@
+        ret = samples[pos];
+    } else {
+        ret = 0;
+        pos -= (step * 3);
+        for (unsigned int i = 0; i < size; ++i) {
+            ret += samples[pos - (3 * i)];
+        }
+
+        ret /= size;
+    }
+
+    return (byte)ret;
+}
+
+/* We need to compress the stream of raw accelerometer data into 128 bytes, so
+ * it will fit into a neuron, while preserving as much of the original pattern
+ * as possible. Assuming there will typically be 1-2 seconds worth of
+ * accelerometer data at 200Hz, we will need to throw away over 90% of it to
+ * meet that goal!
+ *
+ * This is done in 2 ways:
+ *
+ * 1. Each sample consists of 3 signed 16-bit values (one each for X, Y and Z).
+ *    Map each 16 bit value to a range of 0-255 and pack it into a byte,
+ *    cutting sample size in half.
+ *
+ * 2. Undersample. If we are sampling at 200Hz and the button is held for 1.2
+ *    seconds, then we'll have ~240 samples. Since we know now that each
+ *    sample, once compressed, will occupy 3 of our neuron's 128 bytes
+ *    (see #1), then we know we can only fit 42 of those 240 samples into a
+ *    single neuron (128 / 3 = 42.666). So if we take (for example) every 5th
+ *    sample until we have 42, then we should cover most of the sample window
+ *    and have some semblance of the original pattern. */
+void undersample(byte samples[], int numSamples, byte vector[])
+{
+    unsigned int vi = 0;
+    unsigned int si = 0;
+    unsigned int step = numSamples / samplesPerVector;
+    unsigned int remainder = numSamples - (step * samplesPerVector);
+
+    /* Centre sample window */
+    samples += (remainder / 2) * 3;
+    for (unsigned int i = 0; i < samplesPerVector; ++i) {
+        for (unsigned int j = 0; j < 3; ++j) {
+            vector[vi + j] = getAverageSample(samples, numSamples, si + j, step);
+        }
+
+        si += (step * 3);
+        vi += 3;
+    }
+}
+
+void readVectorFromIMU(byte vector[])
+{
+    byte accel[sensorBufSize];
+    int raw[3];
+
+    unsigned int samples = 0;
+    unsigned int i = 0;
+
+    /* Wait until Motion detection inturrupt flag  */
+    while (ledState == false)
+{
+    digitalWrite(11, LOW);  // Optional Beeper
+}
+    /* While Motion detection inturrupt flag being held... */
+    while (ledState == true) {
+
+        if (CurieIMU.accelDataReady()) {
+
+            CurieIMU.readAccelerometer(raw[0], raw[1], raw[2]);
+
+            /* Map raw values to 0-255 */
+            accel[i] = (byte) map(raw[0], IMULow, IMUHigh, 0, 255);
+            accel[i + 1] = (byte) map(raw[1], IMULow, IMUHigh, 0, 255);
+            accel[i + 2] = (byte) map(raw[2], IMULow, IMUHigh, 0, 255);
+
+            i += 3;
+            ++samples;
+
+            /* If there's not enough room left in the buffers
+            * for the next read, then we're done */
+            if (i + 3 > sensorBufSize) {
+                break;
+            }
+
+        }
+digitalWrite(11, HIGH);
+delay(50);
```

Comment:
We are sampling at 200Hz, meaning there will be a new sample for us to read every 5ms.
Busy-waiting for 50ms here at each iteration means you'll likely miss 9 out of every 10 samples.

Not necessarily going to break things, the sketch probably still works as intended, but you're secretly sampling at a much lower resolution than intended. If the LED is needed, can you figure out a way to do it without disrupting the main sensor-reading loop so much?